text/0000-crate-changelog-field.md

```diff
@@ -0,0 +1,276 @@
+
+Current situation: a consumer usually needs to jump through several hops to find the changelog:
+
+```mermaid
+flowchart LR;
+    editor["crates" plugin in vscode]
+    crates-io[crates.io crate page]
+    doc-rs[doc.rs crate page]
+    crate-web[crate website]
+    github[crate repository]
+    google["Google search result"]
+    repo-changelog[changelog in repository]
+    web-changelog[changelog on a website]
+    doc-changelog[changelog on doc.rs]
+    editor --> crates-io
+    editor --> doc-rs
+    crates-io --> github
+    github --> repo-changelog
+    github --> web-changelog
+    github --> doc-changelog
+    google --> crates-io
+    google --> github
+    google --> doc-rs
+    google -->|unlikely| repo-changelog
+    google --> web-changelog
+    google --> doc-changelog
+    google --> crate-web
+    crate-web --> web-changelog
+    crate-web --> repo-changelog
+    crate-web --> doc-changelog
+    doc-rs --> doc-changelog
+    doc-rs --> web-changelog
+    doc-rs --> crates-io
+    doc-rs --> crate-web
+    doc-rs --> github
+```
+
+This topic was brought several times, most notable notions are:
+
+- [Previously closed RFC](https://github.com/rust-lang/rfcs/pull/2129)
+- [Rustaceans: Please Keep a Changelog!](https://blog.dbrgn.ch/2015/12/1/rust-crates-keep-a-changelog/) ([reddit](https://www.reddit.com/r/rust/comments/3v1ndl/))
+- [Cargo issue](https://github.com/rust-lang/cargo/issues/2188)
+- [Reddit thread](https://www.reddit.com/r/rust/comments/6vvhjh/)
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+An optional `changelog` field is available, which is currently in the format
+of a URL to an external resource.
+
+Examples:
+
+```toml
+[package]
+name = "foo"
+version = "0.1.0"
+changelog = "https://github.com/foo-org/foo/releases"
+```
+
+```toml
+[package]
+name = "bar"
+version = "1.0.0"
+changelog = "https://github.com/bar-org/bar/blob/main/CHANGELOG.md"
+```
+
+For crates with `changelog` field, crates.io and docs.rs show a link to the
+changelog in the corresponding page of the crate.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Cargo
+
+A new `changelog` field will be added to crate manifest,
+which only accepts valid URL for now.
+However, it is explicitly stated that in the future the field could be
+extended to handle other formats than URL and such a change is not considered
+breaking.
+So external tools should expect that the field could contain contents that are
+not valid URL and handle it with caution.
+
+For a cargo workspace, it is possible for several crates to share a changelog
+via the usage of `changelog.workspace`, just like
+[`documentation.workspace`](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table).
+
+For now, `cargo publish` will check if the `changelog` field is a valid URL,
+unless `--no-verify` is also specified.
```

Comment:
`--no-metadata` is a better fit.

```suggestion
unless `--no-metadata` is also specified.
```

Reply:
If its a url, shouldn't that be part of the schema and always error, if present?

Reply:
If parsers are supposed to get any meaning from the field, it likely needs one format. Alternative formats could be handled with alternate fields (like license and license-file) or dotted keys (e.g. this could be `changelog.url`)

Reply:
> If parsers are supposed to get any meaning from the field…

Agree with this. It depends on how much functionality we expect from this field.